pkg/controller/machinehealthcheck/machinehealthcheck_controller_test.go

```diff
@@ -327,3 +331,32 @@ func TestUnhealthyForTooLong(t *testing.T) {
 		}
 	}
 }
+
+func TestRemediationOnBareMetalProvider(t *testing.T) {
```

Comment:
Seems like we're missing a positive BM test to see that the annotation is created.

Reply:
+1, just create unhealthy node and verify if after reconcile annotation got added or not.

Reply:
@vikaschoudhary16  Do you know how can I do it? I tried to do it, and at least with the controller-runtime fake client it does not update the node, I can see it enters the update method but the node object that I send to the fake client does not update

Reply:
@cynepco3hahue i can try. will let you know once done.

Reply:
No matter already got it:)